packages/kbn-router-to-openapispec/src/oas_converter/kbn_config_schema/post_process_mutations/mutations/utils.ts

```diff
@@ -35,6 +35,13 @@ export const processDeprecated = (schema: OpenAPIV3.SchemaObject): void => {
   }
 };
 
+export const processDiscontinued = (schema: OpenAPIV3.SchemaObject): void => {
+  if (metaFields.META_FIELD_X_OAS_DISCONTINUED in schema) {
+    schema.discontinued = 'versionOrDate';
```

Comment:
@jloleysens I've hit a blank!! type check doesn't like this...

Reply:
🤞 [`186a3c8`](https://github.com/elastic/kibana/pull/192331/commits/186a3c885ed03fd2758c878ee5e948be43f939f0)

Reply:
I did some reading and realized adding a field to the OpenAPIV3 schema object isn't all that easy. We'd need to extend the base schema, if that's even supported.

Reply:
the pre-commit hook doesn't like kebab-cased file names. I've tried a workaround in [`207d598`)](https://github.com/elastic/kibana/pull/192331/commits/207d598663844b12d39b663fdcdd7ce2f72f9ae4)